### DISCUSSION TITLE
cannot delete indices by query

### DISCUSSION THREAD

**Discussion by bfca667** [Security-Onion-Solutions/securityonion#4067] [Unsupported Versions] [answered]
Hi,
I am new to ELK and SecurityOnion, and what I am trying to do is delete *:so-beats-* indices where event.module field is equal to "auditd", by applying the following curl command:

curl -X POST "localhost:9200/sostorage1_searchnode:so-beats-2021.04.2*/_delete_by_query?pretty" -H 'Content-Type: application/json' -d'
{
  "query": {
    "match": {
      "event.module": "auditd"
    }
  }
}
'

However, I get index_not_found exceptions like this:

 {
      "index" : "so-beats-2021.04.27",
      "type" : "_doc",
      "id" : "OzWqFHkBhIw4renVg_2o",
      "cause" : {
        "type" : "index_not_found_exception",
        "reason" : "no such index [so-beats-2021.04.27]",
        "resource.type" : "index_expression",
        "resource.id" : "so-beats-2021.04.27",
        "index_uuid" : "_na_",
        "index" : "so-beats-2021.04.27"
      },
      "status" : 404
    },

In fact, when I try to list all indices, I cannot even see the indices I am trying to manipulate:

curl -XGET localhost:9200/_cat/indices/

green open elastalert_status_status  7hfki2tsT6WR_L-6RUupmQ 1 0    0   0  208b  208b
green open elastalert_status         katYovtaS9C32iUDUh28dw 1 0    0   0  208b  208b
green open elastalert_status_past    3s2IHEQzQ6uZr1l1HTt2PA 1 0    0   0  208b  208b
green open elastalert_status_silence -KtdY3J6RuavOFTJHjPYYg 1 0    0   0  208b  208b
green open elastalert_status_error   Ygkgp-_BR_e7552eaUCY4A 1 0    0   0  208b  208b
green open .kibana_1                 V2-SBAErQsuxlaveJoNg2w 1 0 2584 132 2.1mb 2.1mb

Thanks for your help

**Comment by weslambert** [comment, marked as answer]
It looks like you are just listing indices from your manager.

If you are running the default Elasticsearch cluster type for Security Onion, you'll want to run the following command on your search nodes:

`curl -sk https://localhost:9200/_cat/indices | sort -n`

To delete an index, try something like the following:

`curl -sk -XDELETE https://localhost:9200/$indexname`

Note that wildcarding indices for deletion is not allowed by default (can be modified in elasticsearch.yml).

If you would like to delete via wildcard without modifying the ES setting (which we don't recommend changing), you'll need to use a `for` loop in doing so.

**Reply by bfca667** [reply]
Thank you for your reply.

Unfortunately, curl -sk https://localhost:9200/_cat/indices | sort -n did not return anything.
Also, when I try to delete an index individually, by issuing curl -sk -XDELETE https://localhost:9200/$indexname, I get the same index_not_found_exception.
But what I really mean to do is not to delete the whole index , but to delete documents returned by a query within that index. These documents should satisfy event.module == auditd, according to my original post.

I appreciate any additional help. Thanks

**Reply by marianoka** [reply]
Hi Wes,
I have run curl -sk https://localhost:9200/_cat/indices | sort -n andI get an auth msg. At 2.3.61 you have to authenticate firts??

"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication credentials for REST request [/_cat/indices]","header":{"WWW-Authenticate":["Basic realm=\"security\" charset=\"UTF-8\"","Bearer realm=\"security\"","ApiKey"]}}],"type":"security_exception","reason":"missing authentication credentials for REST request [/_cat/indices]","header":{"WWW-Authenticate":["Basic realm=\"security\" charset=\"UTF-8\"","Bearer realm=\"security\"","ApiKey"]}},"status":401}

**Reply by marianoka** [reply]
Got the solution
curl -sk https://localhost:9200/_cat/indices -u username:password | sort -n

The you will get all the indices and can delete whatever you want!

**Reply by ashokdev7** [reply]
Hy can you tell me how to open a indicies , getting error

**Reply by dougburks** [reply]
@ashokdev7 From the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and simply provide a hyperlink to the related discussion.
